lib/Search/OpenProjectSearchProvider.php

```diff
@@ -109,13 +109,23 @@ public function search(IUser $user, ISearchQuery $query): SearchResult {
 		$offset = $query->getCursor();
 		$offset = $offset ? intval($offset) : 0;
 		$openprojectUrl = OpenProjectAPIService::sanitizeUrl($this->config->getAppValue(Application::APP_ID, 'openproject_instance_url'));
-		$accessToken = $this->config->getUserValue($user->getUID(), Application::APP_ID, 'token');
+
+		$authorizationMethod = $this->config->getAppValue(Application::APP_ID, 'authorization_method', '');
 		$searchEnabled = $this->config->getUserValue(
 			$user->getUID(),
 			Application::APP_ID, 'search_enabled',
 			$this->config->getAppValue(Application::APP_ID, 'default_enable_unified_search', '0')) === '1';
-		if ($accessToken === '' || !$searchEnabled) {
-			return SearchResult::paginated($this->getName(), [], 0);
+
+		if ($authorizationMethod === OpenProjectAPIService::AUTH_METHOD_OAUTH) {
+			$accessToken = $this->config->getUserValue($user->getUID(), Application::APP_ID, 'token');
+			if ($accessToken === '' || !$searchEnabled) {
+				return SearchResult::paginated($this->getName(), [], 0);
+			}
+		} elseif ($authorizationMethod === OpenProjectAPIService::AUTH_METHOD_OIDC) {
+			$accessToken = $this->service->getOIDCToken();
+			if ($accessToken === null || !$searchEnabled) {
+				return SearchResult::paginated($this->getName(), [], 0);
+			}
```

Comment:
`!accessToken` checks for falsy conditions (including `''`, `null`, `0`). but if we want to be strict then we can have `$accessToken === '' || $accessToken === null`
```suggestion
		if ($authorizationMethod === OpenProjectAPIService::AUTH_METHOD_OAUTH) {
			$accessToken = $this->config->getUserValue($user->getUID(), Application::APP_ID, 'token');
		} elseif ($authorizationMethod === OpenProjectAPIService::AUTH_METHOD_OIDC) {
			$accessToken = $this->service->getOIDCToken();
		}

		if (!$accessToken || !$searchEnabled) {
			return SearchResult::paginated($this->getName(), [], 0);
```